src/environment.ts

```diff
@@ -1,147 +1,52 @@
-    ) {
-      let regexMatch = process.env.MONGO_API_PASS.match(
-        MONGO_ROOT_PASS_REGEX
-      ) as string[];
-      let matchString = regexMatch?.join("");
-      if (matchString !== process.env.MONGO_API_PASS) {
-        throw new Error("MongoDB API password not complex enough!");
-      }
-    }
+const secrets: Record<string, [string | undefined, RegExp]> = {
+  JWT_SECRET: [process.env.JWT_SECRET, JWT_SECRET_REGEX]
+};
 
-    // Validate MongoDB Root Password
-    if (!process.env.MONGO_ROOT_PASS) {
-      throw new Error("No MongoDB root password set!");
-    } else if (
-      Environment.PRODUCTION === process.env.WESTEGG_ENV ||
-      Environment.STAGING === process.env.WESTEGG_ENV
-    ) {
-      let regexMatch = process.env.MONGO_ROOT_PASS.match(
-        MONGO_ROOT_PASS_REGEX
-      ) as string[];
-      let matchString = regexMatch?.join("");
-      if (matchString !== process.env.MONGO_ROOT_PASS) {
-        throw new Error("MongoDB root password not complex enough!");
-      }
-    }
+if (!Object.values(required).every(Boolean)) {
+  const variable = Object.entries(required).find(([key, value]) => !value)?.[0];
+  throw new Error(`Missing environment variable: ${variable}`);
+}
 
-    if (!process.env.MONGO_API_USER) {
-      throw new Error("MongoDB api user missing!");
-    } else {
-      let regexMatch = process.env.MONGO_API_USER.match(
-        MONGO_API_USER_REGEX
-      ) as string[];
-      let matchString = regexMatch?.join("");
-      if (matchString !== process.env.MONGO_API_USER) {
-        throw new Error("MongoDB api user malformed!");
-      }
-    }
+function validate() {
+  const variable = Object.entries(secrets).find(
+    ([key, [value, regex]]) => !regex.test(value as string)
+  )?.[0];
 
-    // Validate IPFS host url
-    if (
-      Environment.PRODUCTION === process.env.WESTEGG_ENV ||
-      Environment.STAGING === process.env.WESTEGG_ENV
-    ) {
-      let regexMatch = process.env.IPFS_URL?.match(IPFS_URL_REGEX) as string[];
-      let matchString = regexMatch?.join("");
-      if (matchString !== process.env.IPFS_URL || !process.env.IPFS_URL) {
-        throw new Error("IPFS host missing or malformed!");
-      }
-    }
+  if (variable) {
+    throw new Error(`Secret key is missing required complexity: ${variable}`);
+  }
+}
 
-    // Validate Sendgrid API key
-    if (!process.env.SENDGRID_API_KEY) {
-      throw new Error("SendGrid API key missing!");
-    } else if (
-      Environment.PRODUCTION === process.env.WESTEGG_ENV ||
-      Environment.STAGING === process.env.WESTEGG_ENV
-    ) {
-      let regexMatch = process.env.SENDGRID_API_KEY.match(
-        SENDGRID_API_KEY_REGEX
-      ) as string[];
-      let matchString = regexMatch?.join("");
-      if (matchString !== process.env.SENDGRID_API_KEY) {
-        throw new Error("SendGrid API key malformed!");
-      }
-    }
+if (!isDevelopment) {
+  try {
+    validate();
   } catch (error) {
-    logger.error(`FATAL: ${error}`);
```

Comment:
Why remove FATAL here? This is indicating the error is stopping westegg, important to specify I would think.

Reply:
Because all unhandled errors should be fatal and the error message already prepends the string `Error:` to the message.

Reply:
And also, text is printed when the program exits with an error anyway.